pkg/cmd/cluster/cluster_link.go

```diff
@@ -0,0 +1,129 @@
+package cluster
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/url"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/dcos/dcos-cli/pkg/config"
+	"github.com/dcos/dcos-cli/pkg/login"
+	"github.com/dcos/dcos-cli/pkg/setup"
+	"github.com/spf13/cobra"
+)
+
+// newCmdClusterLink links the attached cluster to another one.
+func newCmdClusterLink(ctx api.Context) *cobra.Command {
+	setupFlags := setup.NewFlags(ctx.Fs(), ctx.EnvLookup)
+	cmd := &cobra.Command{
+		Use:  "link",
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			attachedCluster, err := ctx.Cluster()
+			if err != nil {
+				return err
+			}
+
+			manager := ctx.ConfigManager()
+			linkableClusterConfig, err := manager.Find(args[0], false)
+			if err != nil {
+				return err
+			}
+
+			if config.Empty() == linkableClusterConfig {
+				// The cluster does not exist yet. Two possibilities:
+				// - The argument is an URL, we try to setup the cluster.
+				// - The argument is not an URL, we return an error.
+				_, err = url.ParseRequestURI(args[0])
```

Comment:
Does this support passing an URL without scheme? Currently we accept such arguments and assume `https://`. We should also move this validation logic to the Setup function not to duplicate the logic with the setup command.